eZ/Publish/API/Repository/Values/VersionState/VersionState.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * File containing the VersionState class
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace eZ\Publish\API\Repository\Values\VersionState;
+
+use eZ\Publish\API\Repository\Values\ValueObject;
+
+/**
+ * This class represents a version state value
+ *
+ * @property-read mixed $id the id of the content type group
+ * @property-read string $identifier the identifier of the content type group
+ * @property-read string $defaultLanguageCode the default language of the version state group names and description used for fallback.
+ * @property-read string[] $languageCodes the available languages
+ */
+abstract class VersionState extends ValueObject
+{
+    /**
+     * Primary key
+     *
+     * @var mixed
+     */
+    protected $id;
+
+    /**
+     * Readable string identifier of the version state
+     *
+     * @var string
+     */
+    protected $identifier;
+
+    /**
+     * Priority for ordering
+     *
+     * @var int
+     */
+    protected $priority;
```

Comment:
yes, removed it for now. But as we talked about it might make sense to keep it despite skipping groups. I'll try to discuss this with @paulborgermans which where involved in the original object states work.